mcstatus/utils.py

```diff
@@ -49,31 +50,60 @@ def sync_wrapper(*args, tries: int = tries, **kwargs):
     return decorate
 
 
-def deprecated(replacement: str = None, version: str = None, msg: str = None):
-    def decorate(func: Callable) -> Callable:
-        # Construct and send the warning message
-        name = getattr(func, "__qualname__", func.__name__)
-        warn_message = f"'{name}' is deprecated and will be removed"
-        if version is not None:
-            warn_message += f" in {version}"
-        if replacement is not None:
-            warn_message += f", use '{replacement}' instead"
-        warn_message += "."
-        if msg is not None:
-            warn_message += f" ({msg})"
+def deprecated(
+    *,
+    replacement: Optional[str] = None,
+    version: Optional[str] = None,
+    date: Optional[str] = None,
+    msg: Optional[str] = None,
+    methods: Optional[Iterable[str]] = None,
+):
+    if date is not None and version is not None:
+        raise ValueError("Expected removal timeframe can either be a date, or a version, not both.")
 
+    def decorate_func(func: Callable, warn_message: str):
         @wraps(func)
         def sync_wrapper(*args, **kwargs):
-            warnings.warn(warn_message)
+            warnings.warn(warn_message, category=DeprecationWarning)
             return func(*args, **kwargs)
 
         @wraps(func)
         async def async_wrapper(*args, **kwargs):
-            warnings.warn(warn_message)
+            warnings.warn(warn_message, category=DeprecationWarning)
             return await func(*args, **kwargs)
 
         if asyncio.iscoroutinefunction(func):
             return async_wrapper
         return sync_wrapper
 
+    def decorate(obj: Callable) -> Callable:
+        # Construct and send the warning message
+        name = getattr(obj, "__qualname__", obj.__name__)
+        warn_message = f"'{name}' is deprecated and is expected to be removed"
+        if version is not None:
+            warn_message += f" in {version}"
+        elif date is not None:
+            warn_message += f" in {date}"
```

Comment:
Would to be removed in vX on YYYY-MM read better?
```suggestion
            warn_message += f" on {date}"
```